keyboards/gopolar/gg86/keymaps/default/keymap.c

```diff
@@ -0,0 +1,314 @@
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                        rgb_matrix_enable_noeeprom();
+                        user_config.rgb_mode = RGB_MODE_ALL;
+                    }
+                    break;
+                }
+                eeconfig_update_user(user_config.raw);
+            }
+            return false;
+	}
+
+    return true;
+}
+
+#ifdef OLED_ENABLE
+    uint16_t startup_timer; 
+
+    oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+        startup_timer = timer_read();
+        wave_timer = timer_read();
+
+        return rotation;
+    }
+
+    static void render_layer_info(void) {
+        switch (get_highest_layer(layer_state)) {
+            case 0:
+                oled_write_P(PSTR("QWERTY "), false);
+                break;
+            case 1:
+                oled_write_P(PSTR("Function "), false);
+                break;
+            case 2:
+                oled_write_P(PSTR("Function "), false);
+                break;
+            case 3:
+                oled_write_P(PSTR("Function "), false);
+                break;
+            default:
+                oled_write_P(PSTR("Undefined "), false);
+        }
+    }
+
+    static void render_layer_number(void) {
+        switch (get_highest_layer(layer_state)) {
+            case 0:
+                oled_write_P(PSTR("0  "), false);
+                break;
+            case 1:
+                oled_write_P(PSTR("1"), false);
+                break;
+            case 2:
+                oled_write_P(PSTR("2"), false);
+                break;
+            case 3:
+                oled_write_P(PSTR("3"), false);
+                break;
+            default:
+                oled_write_P(PSTR("X  "), false);
+        }
+    }
+
+    bool oled_task_user(void) {
+        static bool finished_logo = false;
+
+        if ((timer_elapsed(startup_timer) < 5000) && !finished_logo) {
+            render_logo();
+        } else {
+            finished_logo = true;
+
+            led_t led_usb_state = host_keyboard_led_state();
+
+            oled_write_P(led_usb_state.caps_lock ? PSTR("CAPLK ") : PSTR("----- "), false);
+            oled_write_P(led_usb_state.num_lock ? PSTR(" NUMLK ") : PSTR(" ----- "), false);
+            oled_write_P(led_usb_state.scroll_lock ? PSTR(" SCRLK") : PSTR(" -----"), false);
+
+            // sleep if it has been long enough since we last got a char
+            if (timer_elapsed32(wave_sleep) > OLED_TIMEOUT) {
+                oled_off();
+            } else {
+                oled_on();
+            }
+            // time for the next frame?
+            if (timer_elapsed(wave_timer) > FRAME_TIMEOUT) {
+                wave_timer = timer_read();
+                render_frame();
+            }
+
+            oled_set_cursor(0, 3);
+            render_layer_info();
+            oled_write_P(PSTR("Layer: "), false);
+            render_layer_number();
+        }
+
+        return true;
+    }
+#endif
```

Comment:
```suggestion
```

This should be moved to the keyboard's c file (gg86.c), and use the `kb` version of the functions.  That way, a default behavior is set, and can be overridden by users.

Reply:
Try moving them on gg86.c, after change to `kb` version. I can't use `user` version for my keymap or bongocat

Reply:
That probably is missing the user check.

Reply:
Done, i have to keep function for wave keylogger on keymap. Because it needed to use with process_record_user